modules/gdscript/gdscript_vm.cpp

```diff
@@ -1666,7 +1680,13 @@ Variant GDScriptFunction::call(GDScriptInstance *p_instance, const Variant **p_a
 					base->callp(*methodname, (const Variant **)argptrs, argc, ret, err);
 				}
 #ifdef DEBUG_ENABLED
-				if (GDScriptLanguage::get_singleton()->profiling) {
+				// In some cases, non-verified builtin functions end up in OPCODE_CALL.
+				// When that happens, base_class is empty, so we can use this to stop
+				// the time from being subtracted in that case
+				//
```

Comment:
I don't see why this would be the case. They should still have a base Object which will always have a class name (unless they are freed or it's `null`). The only exception are calls on built-in types (Vector2, String, etc.) which don't have a base Object.

Reply:
You're right, I just completely missed it. Then, what's a good way to distinguish the two at that point? Would the base Object's get_script work? Native calls shouldn't have scripts attached to them, right?

Other than that I can only think of changing the compiler and opcodes, which seems inadvisable.

Reply:
You're right, that was my mistake. I can't figure out a nice way to distinguish native from user GDScript  code at this point in the VM without bigger changes. After thinking about it for a while and testing some things, the best I could find was:
```cpp
if (!base_obj || !base_obj->get_script_instance() || !base_obj->get_script_instance()->has_method(*methodname))
```
 But I don't know, there seem to be a ton of potential corner cases. Can you think of a scenario where this would fail? Or a decent alternative?  I'll spend some more time testing this more thoroughly regardless.